src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -59,63 +57,104 @@ final class SettingsTable extends AbstractVirtualTable
                            .addPartitionKeyColumn(NAME, UTF8Type.instance)
                            .addRegularColumn(VALUE, UTF8Type.instance)
                            .build());
-        this.config = config;
+        for (String name : BACKWARDS_COMPATABLE_NAMES.keySet())
+        {
+            if (DatabaseDescriptor.hasProperty(name))
+            {
+                throw new AssertionError(String.format("Name '%s' is present in Config, this adds a conflict as this " +
+                                                       "name had a different meaning in the '%s",
+                                                       name, SettingsTable.class.getSimpleName()));
+            }
+        }
+    }
+
+    @Override
+    protected void applyColumnUpdate(AbstractMutableVirtualTable.ColumnValues partitionKey,
```

Comment:
I would add a new column into this vtable called "mutable" of type `boolean`. That way, a user would see what properties are mutable and what are not. Currently, I do not see this instantly, do I? How do I know what I am able to change? 

So if some specific column is not mutable, I would just make the call to this method no-op and we might just focus on the logic where we indeed going to update something.

Reply:
I tend to agree that this column must appear, but this raises the question of backward compatibility of the virtual table if we add a new column: "Are we able to add new columns to VT? How do users use VTs?".  As these changes require a lot of discussion in themselves and require collecting user experience with VTs, I have left them to be followed up.

So my plan is:

- prepare and merge the initial changes;
- start a discussion with VT about backwards compatibility and then add an extra column to the SettingsTable (or as an option create SettingsTableV2 in the worst case);
- migrate other properties that should be updatable as well;

The reason for not including the column in this PR is to minimize the changes and make them less debatable.
WDYT?

Reply:
As you wish but I dont see that to be problematic. VT are just local tables ... there is also implicit `allow filtering` on them ... it is not part of primary key so whats the big deal? 5.0 is ideal place to put this in. When do we want to do that, in a patch release?

Reply:
no SettingsV2, no no .... unnecessary, we can not do new versions of a table and duplicate it all just because we added a column

Reply:
I was targeting this for 5.0 and 4.1 releases, so I tried to avoid any public API changes. For that case, we can cherry-pick the changes that make SettingsTable updatable for 4.1, leaving the column addition for 5.0 only, but as we made `allow filtering` the default for vtable (I forgot about it) I don't see much difference now.

@dcapwell what do you think about adding a new column?

Reply:
I was thinking this goes only to 5.0. Why are you targeting 4.1 too?

Reply:
To improve cluster manageability in general, I thought that since the 4.x has an EOL until mid-2025 and the 5.0 stabilisation takes at least a few months as I understand the whole process, I thought it would be great to get users used to with managing their cluster with tables before they switch to the 5.0.

The feature itself is big enough but not too big as other CEPs.

Reply:
Rumor has it that I think the freeze for 5.0 should be done around August so we should hurry up if you want to see this in 5.0 too. Another option would be to put it to 4.1 and then in 5.0 we would just add that column as a "breaking change".

I can not answer what to do with 4.1 though (if it should be there, this whole feature), would you mind to drop an email to ML?

Reply:
Sure, but there is a lack of feedback on this :-) we can leave it for now and come back to it when everything else is ready for the merge.